### ISSUE TITLE
[NBug] fatal: bad object 2222222222222222222222222222222222222222

### ISSUE THREAD

**Issue by velteyn** [gitextensions/gitextensions#11388]
### Environment

- Git Extensions 4.2.0.17591
- Build bf814d4f1c6f178eacef38d592b700f9f2519dcb
- Git 2.43.0.windows.1
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.24
- DPI 120dpi (125% scaling)
- Portable: False

### Issue description

Exit code: 128
Command: C:\Program Files\Git\bin\git.exe
Arguments: -c log.showSignature=false log --pretty="format:%G?" -1 2222222222222222222222222222222222222222
Working directory: C:\Devel\git-repos\sfm\

```
GitExtUtils.ExternalOperationException: fatal: bad object 2222222222222222222222222222222222222222
 ---> System.Exception: fatal: bad object 2222222222222222222222222222222222222222
   --- End of inner exception stack trace ---
   at async Task<string> GitCommands.ExecutableExtensions.GetOutputAsync(IExecutable executable, ArgumentString arguments, byte[] input, Encoding outputEncoding, CommandCache cache, bool stripAnsiEscapeCodes) in C:/projects/gitextensions/GitCommands/Git/ExecutableExtensions.cs:line 147
   at void Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at T Microsoft.VisualStudio.Threading.JoinableTask<T>.CompleteOnCurrentThread()
   at T Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run<T>(Func<Task<T>> asyncMethod, JoinableTaskCreationOptions creationOptions) x 2
   at string GitCommands.ExecutableExtensions.GetOutput(IExecutable executable, ArgumentString arguments, byte[] input, Encoding outputEncoding, CommandCache cache, bool stripAnsiEscapeCodes) in C:/projects/gitextensions/GitCommands/Git/ExecutableExtensions.cs:line 43
   at Task<CommitStatus> GitCommands.Gpg.GitGpgController.GetRevisionCommitSignatureStatusAsync(GitRevision revision)+() => { } in C:/projects/gitextensions/GitCommands/Git/GitGpgController.cs:line 108
   at async Task<CommitStatus> GitCommands.Gpg.GitGpgController.GetRevisionCommitSignatureStatusAsync(GitRevision revision) in C:/projects/gitextensions/GitCommands/Git/GitGpgController.cs:line 99
   at async Task<GpgInfo> GitUI.CommandsDialogs.FormBrowseController.LoadGpgInfoAsync(GitRevision revision) in C:/projects/gitextensions/GitUI/CommandsDialogs/FormBrowseController.cs:line 32
   at async Task GitUI.CommandsDialogs.FormBrowse.FillGpgInfoAsync(GitRevision revision) in C:/projects/gitextensions/GitUI/CommandsDialogs/FormBrowse.cs:line 1276
   at async void GitUI.TaskManager.InvokeAndForget(Control control, Func<Task> asyncAction, CancellationToken cancellationToken)+(?) => { } [1] in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 111
   at async Task GitUI.TaskManager.HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync) in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 38
```

dunno !! Maybe a type by the cat ?

### Steps to reproduce

the cat passed on the keyboard

**Comment by pmiossec** [comment]
>the cat passed on the keyboard

Can you provide us your cat so that we can reproduce? 🤣

**Comment by velteyn** [comment]
🤣

Il giorno ven 24 nov 2023 alle ore 09:33 Philippe Miossec <
***@***.***> ha scritto:

**Comment by bansan85** [comment]
Same backtrace for me :

Exception throws when I clicked on "Working directory" line.

No reproducible.

`fatal: bad object 1111111111111111111111111111111111111111`


Exit code: 128
Command: C:\Program Files\Git\bin\git.exe
Arguments: -c log.showSignature=false log --pretty="format:%G?" -1 1111111111111111111111111111111111111111
Working directory: C:\Users\v.legarrec\Documents\git\urx.wiki\

```
GitExtUtils.ExternalOperationException: fatal: bad object 1111111111111111111111111111111111111111
 ---> System.Exception: fatal: bad object 1111111111111111111111111111111111111111
   --- End of inner exception stack trace ---
   at async Task<string> GitCommands.ExecutableExtensions.GetOutputAsync(IExecutable executable, ArgumentString arguments, byte[] input, Encoding outputEncoding, CommandCache cache, bool stripAnsiEscapeCodes) in C:/projects/gitextensions/GitCommands/Git/ExecutableExtensions.cs:line 147
   at void Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at T Microsoft.VisualStudio.Threading.JoinableTask<T>.CompleteOnCurrentThread()
   at string GitCommands.ExecutableExtensions.GetOutput(IExecutable executable, ArgumentString arguments, byte[] input, Encoding outputEncoding, CommandCache cache, bool stripAnsiEscapeCodes) in C:/projects/gitextensions/GitCommands/Git/ExecutableExtensions.cs:line 43
   at Task<CommitStatus> GitCommands.Gpg.GitGpgController.GetRevisionCommitSignatureStatusAsync(GitRevision revision)+() => { } in C:/projects/gitextensions/GitCommands/Git/GitGpgController.cs:line 108
   at async Task<CommitStatus> GitCommands.Gpg.GitGpgController.GetRevisionCommitSignatureStatusAsync(GitRevision revision) in C:/projects/gitextensions/GitCommands/Git/GitGpgController.cs:line 99
   at async Task<GpgInfo> GitUI.CommandsDialogs.FormBrowseController.LoadGpgInfoAsync(GitRevision revision) in C:/projects/gitextensions/GitUI/CommandsDialogs/FormBrowseController.cs:line 32
   at async Task GitUI.CommandsDialogs.FormBrowse.FillGpgInfoAsync(GitRevision revision) in C:/projects/gitextensions/GitUI/CommandsDialogs/FormBrowse.cs:line 1276
   at async void GitUI.TaskManager.InvokeAndForget(Control control, Func<Task> asyncAction, CancellationToken cancellationToken)+(?) => { } [1] in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 111
   at async Task GitUI.TaskManager.HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync) in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 38
```

```
- Git Extensions 4.2.0.17591
- Build bf814d4f1c6f178eacef38d592b700f9f2519dcb
- Git 2.42.0.windows.2
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.25
- DPI 96dpi (no scaling)
- Portable: False
```

**Comment by pmiossec** [comment]
Ok, I got it. It's when clicking or having "GPG" tab opened and selecting "Working directory" or "Commit index" pseudo commits.

**Comment by pmiossec** [comment]
PR to fix the issue done: #11389
